### ISSUE TITLE
Non existing member does not get reported when using +=, -=,... operators

### ISSUE THREAD

**Issue by meollab** [microsoft/pylance-release#1858]
## Environment data

* Language Server version: 2021.9.2
* OS and version: Linux x64
* Python version (and distribution if applicable, e.g. Anaconda): Python 3.7.5 (64bit)
* python.analysis.indexing: undefined
* python.analysis.typeCheckingMode: basic

## Expected behaviour

```python
class ExampleA:
    def blah(self) -> None:
        self.does_not_exist_a += 3
```
should report `does_not_exist_a` as not existing 

## Actual behaviour
Pyright does not report any error or warning

## Code Snippet / Additional information
* The problem with `does_not_exist_a` is reported by both mypy and pylint
    * mypy: `error: "ExampleA" has no attribute "does_not_exist_a"`
    * pylint: `E1101: Instance of 'ExampleA' has no 'does_not_exist_a' member (no-member)`
* Pyright correctly reports `Cannot access member "does_not_exist_b" for type "ExampleB"  Member "does_not_exist_b" is unknown` here
```python
class ExampleB:
    def blah(self) -> None:
        print(self.does_not_exist_b)
```

**Comment by erictraut** [comment]
Thanks for the bug report. I agree that this is a false negative case that a static type checker can detect. In particular, if the only assignments to an instance variable are through augmented assignment expressions, it should be flagged as an error.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202193-22-september-2021
